### DISCUSSION TITLE
FIDO2 2FA not working when using reverse proxy

### DISCUSSION THREAD

**Discussion by Anonym234** [dani-garcia/vaultwarden#3014] [Q&A]
I use the vaultwarden/server docker container behind a nginx-based reverse proxy. The connection between nginx and the vaultwarden docker container is via http (nginx redirects to `http://localhost:<some port>`) and my docker-compose config redirects port <some port> to port 80 in the vaultwarden container.

When I log in, the browser extension or website asks for my 2FA and displays an error message, saying that this operation is not allowed.

**Comment by BlackDex** [comment]
Then your nginx added or modified some security headers which prevents a good working environment.

Headers like `Content-Security-Policy` or `X-Frame-Options` should not be altered, or in some cases not even sent, like for the popup screens used for mfa, because, else they are blocked.

Also, the DOMAIN needs to be configured correctly, else that will cause issues too.
And you need https if it is not `localhost` or `127.0.0.1`.
